src/device/nxp/hardfault.go

```diff
@@ -0,0 +1,29 @@
+// Hand created file. DO NOT DELETE.
+// Hardfault aliases for definitions that have inconsistent naming (which are
+// auto-generated by gen-device-svd.go) among devices in package nxp.
+
+// +build nxp,!mimxrt1062
+
+package nxp
+
+const (
+	HardFault_CFSR_IACCVIOL    = SystemControl_CFSR_IACCVIOL
```

Comment:
It appears that these values are not specific to NXP chips but instead for the whole Cortex-M7 family. Take a look here:
https://developer.arm.com/documentation/dui0646/b/cortex-m7-peripherals/system-control-block/configurable-fault-status-register?lang=en

I would suggest moving this into src/device/arm/arm.go (which implies hand-defining them, but that only needs to be done once). That also avoids NXP-specific code in src/runtime/runtime_cortexm_hardfault_debug.go.

Reply:
The values weren't the problem here, but rather the identifiers that were generated from the SVD. The values actually apply to several Cortex-M families (not just the M7).

For some reason, the NXP MIMXRT1062 SVD used `SystemControl_` prefix, while other Cortex-M devices used `SCB_`. The existing code trying to reference `SCB_` fields thus failed to compile for this chip (see: `runtime/runtime_cortexm_hardfault_debug.go`), even though the values did exist (as you noted), they just needed to use correct identifiers with a different prefix. 

So I don't think it would be appropriate to move these definitions to a more global scope, since most Cortex-M devices do not use `SystemControl_`-prefixed identifiers for these field values.

Hope that makes sense and doesn't miss the point you're making!

Reply:
What I suggest is to ignore the identifiers in the SVD file entirely and define them manually, as they are NXP specific and even vary by chip (apparently) while their values are the same across all chips (M7 or maybe lower). In fact, most SVD files do not contain definitions for these peripherals that are defined by ARM.

I know that these identifiers are used in `runtime/runtime_cortexm_hardfault_debug.go`, and that's the problem. A generic Cortex-M runtime file should not import a NXP device package, that's a bug that I apparently missed previously. So basically, `runtime/runtime_cortexm_hardfault_debug.go` should be modified to import device/arm instead of device/nxp.

Reply:
The routines `GetFaultStatus()`, `(MemFaultStatus) Address()`, and `(BusFaultStatus) Address()` all refer to a `nxp.SystemControl` block of registers. 

The `SCB_Type` in package `device/arm` doesn't have the registers they are referencing.

Besides manually re-defining the original fields in, say, `device/arm/faultstatus.go`, it seems the `CFSR`, `MMFAR`, and `BFAR` registers will all also need to be manually defined here as well.

Were you expecting these changes with your original suggestion? Or should this be moved to its own PR?

Reply:
> The routines `GetFaultStatus()`, `(MemFaultStatus) Address()`, and `(BusFaultStatus) Address()` all refer to a `nxp.SystemControl` block of registers.

Yes, and that's a problem. This should be changed to not import device/nxp and import device/arm instead.

> The `SCB_Type` in package `device/arm` doesn't have the registers they are referencing.

Yes, they should be added to `SCB_Type`.

> Besides manually re-defining the original fields in, say, `device/arm/faultstatus.go`, it seems the `CFSR`, `MMFAR`, and `BFAR` registers will all also need to be manually defined here as well.

What do you mean re-defining? It is true that these registers will need to be written by hand in the device/arm package, but only once.

Reply:
I mean "re-defining" in the sense that several (many?) devices, such as these `device/nxp` implementations, already have these fields defined in their SVD auto-generated files. So, for these cases, we are duplicating the field and register definitions by also adding them to package `device/arm`. 

I _didn't_ mean "re-define" as in renaming/moving or changing the values in those existing generated packages.

Reply:
I wasn't sure if changing the type definition of `SCB_Type` in `device/arm` was warranted for this PR. 

But I will go ahead and make these changes and push in a few hours

Reply:
Marking this conversation resolved, see 4091bca for all requested changes.